asyncwrapper/src/main/kotlin/org/n/riesgos/asyncwrapper/dummy/AbstractWrapper.kt

```diff
@@ -432,7 +434,13 @@ abstract class AbstractWrapper(val publisher : PulsarPublisher, val wpsConfigura
         // Now, we know that we haven't found any existing link for it.
         // so we are going to upload it.
         val fileStorage = FileStorage(filestorageConfig.endpoint, filestorageConfig.user, filestorageConfig.password)
-        fileStorage.upload(filestorageConfig.bucketName, checksum, content, mimeType)
+        retry(wpsConfiguration.retryConfiguration.maxRetries, wpsConfiguration.retryConfiguration.backoffMillis, { ex ->
+            ex is IOException || ex is ServerException || ex is InternalException
```

Comment:
Are these all relevant, recoverable exceptions?

Reply:
Honestly, I didn't think a lot about it so far.

Maybe we can tackle it down.
As the java file that I needed to use (due to the `object` method of the minio client) I was forced to list all the exceptions that the upload code can throw (not regarding the runtime exceptions - but I think we don't need to handle those):

| Exception class | Retry could help | Comment |
|-----------------------|-----------------------|--------------|
| IOException | | |
| InvalidKeyException | | |
| InvalidResponseException | | |
| InsufficientDataException | | |
| NoSuchAlgorithmException | | |
| ServerException | | |
| InternalException | | |
| XmlParserException | | |
| ErrorResponseException | | |

(Currently work-in-progress).

Reply:
So the exceptions I would catch for the retry are:
- IOException
- XmlParserException
- InvalidResponseException

Reply:
@arnevogt Do you think those make sense?

Reply:
@nbrinckm Yes, it makes sense.
I think we can add ErrorResponseException as well just to be sure. Even if it creates unnecessary retries it would not hurt to much.